src/Engine/Path.java

```diff
@@ -164,14 +163,18 @@ public int getPathLength()
   /**
    * @return the amount of fuel it would cost to travel this path with the given unit type 
   **/
-  public int getFuelCost(UnitModel model, GameMap map)
+  public int getFuelCost(Unit unit, GameMap map)
```

Comment:
Can we change `FloodFillFunctor` to operate on a `UnitModel` instead of needing an actual unit? Maybe have an optional `movePower` param int the constructor to make it place nice. I don't like requiring information that isn't needed for the task being performed, which in this case (and the other one we are bickering over) causes unneeded complication.
I know you want FFF to be *the* way that moves are evaluated, but this is another case where the old formulation looks much more straightforward.

Reply:
>`UnitModel`
No. It needs to know the affiliation of the unit, so it can be blocked by opposing forces.
In addition, it needs to know the *exact* unit that is moving, so it can always allow ending on that unit (regardless of whether you're allowing it to end on *other* units).

>`movePower`
Not a bad idea; I guess it'd just be used for an overload that doesn't take `initialFillPower`?
Removing it from the function inputs entirely would be annoying since `Utils` is using it.

>old fomulation is more straightforward
It wouldn't be unreasonable to do a cost-output function.

Reply:
> No. It needs to know... In addition, it needs to know

That's my point though. `getFuelCost` *doesn't* need to know any of that stuff. You are placing more constraints on this function and making it harder to use (as evidenced by the test file changes) for no reason other than that you want to use FFF to accomplish the task. That's enough reason for me to object even if the change didn't make the function more verbose and harder to read as well.
I get that you may want to have a move-cost function that requires a `from`/`to` pair so we can disallow certain transitions (at least, that's the only reason I can think of to not just have a `to` field), but this just feels like a case of "I have a new hammer, and what are screws?".

Reply:
> You are placing more constraints on this function and making it harder to use (as evidenced by the test file changes) for no reason other than that you want to use FFF to accomplish the task.

That's accurate, but I want to use FFF because to do otherwise could generate inconsistent results.
Let's say I add a hovercraft unit, which can go from `SEA` to `GRASS` at a cost of 4 move (but otherwise moves at 1 cost on either). The *actual* fuel cost would be drastically lower than it should be, since this function would be used to calculate it, in a world where FFF was not used.

> I get that you may want to have a move-cost function that requires a from/to pair so we can disallow certain transitions (at least, that's the only reason I can think of to not just have a to field), but this just feels like a case of "I have a new hammer, and what are screws?".

I suppose part of it is that I've had this sort of capability gnawing at the back of my brain for longer than we've even *had* this game.
As for "I have a new hammer, and what are screws?", I would argue the screws are actually functionally inappropriate for what I have planned.

Reply:
>  because to do otherwise could generate inconsistent results

And that's fair. But I should not need a hovercraft to be able to ask "How much fuel does it take for a hovercraft to climb from `SEA` to `GRASS`? The fact that we *do* need one now feels like a regression.

> I would argue the screws are actually functionally inappropriate for what I have planned.

Aye, and I don't doubt that's true. But hammers and nails aren't called for here.

Reply:
> And that's fair. But I should not need a hovercraft to be able to ask "How much fuel does it take for a hovercraft to climb from `SEA` to `GRASS`? The fact that we _do_ need one now feels like a regression.

What alternative would you propose? I guess I could add null checks so it doesn't *need* the unit field, so you could leave it null if you want?
My intent is to centralize control and knowledge of where a unit can go and how, to the greatest feasible extent.

Reply:
> What alternative would you propose?

I would change the `getFuelCost` function to take a `UnitModel` again.

> Ninjas 

Sure, and I'm sure we could come up with yet more additional possible features that would require more feature creep (e.g. Pirates who can board/commandeer enemy boats, which could require the ability to end a move on an enemy). Right now though, we don't have that use case, and when (if) we get there we could easily add an overload that takes map residents and traveler allegiance into account. I don't want to raise the barrier to asking simple questions.